lib/misc.c

```diff
@@ -1282,3 +1282,38 @@ gftp_get_share_dir (void)
   return (gftp_share_dir);
 }
 
+char *
+gftp_format_file_size(off_t bytes) {
+  char tmp[20];
+  char ret[40];
+  char *pointzero = 0;
+  off_t TB = 1024LL * 1024LL * 1024LL * 1024LL;
+  off_t GB = 1024LL * 1024LL * 1024LL;
+  off_t MB = 1024LL * 1024LL;
+  if (bytes >= TB) {
+    snprintf(tmp, sizeof(tmp), "%.2f", (double) bytes / TB);
+    pointzero = strstr(tmp, ".00");
+    if (pointzero) *pointzero = 0;
+    snprintf(ret, sizeof(ret), "%s TiB", tmp);
```

Comment:
`ret` is allocated on the stack and the pointer will no longer be valid once the function returns. This will lead to memory corruption. One way to fix this is to add two parameters to gftp_format_file_size() for the character pointer and the size, The character array would then be defined in report_list_info().

Reply:
Ohhh... I see the g_strdup() below... nevermind. I would still consider passing in a character buffer and length as a function argument so that you avoid moving the data around multiple times.